### PR TITLE
Fixed Misleading Typo in CHANGELOG.md related to false solidity version

### PR DESCRIPTION
**Pull request by 0xprinc**
Fixes #???? 

#### PR Checklist

- [ ] Tests
- [x] Documentation
- [ ] Changeset entry (run `npx changeset add`)

### CODE REVIEW

**Comment by Amxx** [comment]
Hello @0xprinc 

I don't understand this requested change. All files in the release branch use `pragma solidity ^0.8.20`.

The only file in our codebase that doesn't use this pragma is `contracts/mocks/governance/GovernorStorageMock.sol`. That is a mock file, that is not released and should have no effect on the user.

**Comment by 0xprinc** [comment]
Hello sir, 
Actually there is a typo that I have reported. 
Its written 
`Bumped minimum compiler version required to 0.8.20`
But instead it should be `0.8.19` as the link to the pull request this hyperlink is mentioning to is 
`Bump minimum pragma version to 0.8.19`

Actually this typo got me to get false insights about the smart contract I was doing security analysis of.

**Comment by Amxx** [comment]
I think there is a missunderstanding here. 

Yes, the PR #4288 did update the minimum version to 0.8.19 ... but then we did [another PR](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4288) that changed that to 0.8.20.
We decided to not have 2 CHANGELOG entries, to avoid confusion ... and instead we updated the entry that was already there.

Maybe we should change the PR link to the latest one (the one that targets 0.8.20).

File: CHANGELOG.md [hunk 1]
```diff
@@ -48,7 +48,7 @@ These removals were implemented in the following PRs: [#3637](https://github.com
 #### General
 
 - Replaced revert strings and require statements with custom errors. ([#4261](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4261))
-- Bumped minimum compiler version required to 0.8.20 ([#4288](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4288))
+- Bumped minimum compiler version required to 0.8.19 ([#4288](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4288))
```
Comment (Amxx) [inline CHANGELOG.md:51]:
```suggestion
- Bumped minimum compiler version required to 0.8.20 ([#4489](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4489))
```

Comment (ernestognw) [reply CHANGELOG.md]: We should reference both PRs in my opinion.

```suggestion
- Bumped minimum compiler version required to 0.8.20 ([#4489](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4489)) ([#4288](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4288))
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -48,7 +48,7 @@ These removals were implemented in the following PRs: [#3637](https://github.com
 #### General
 
 - Replaced revert strings and require statements with custom errors. ([#4261](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4261))
-- Bumped minimum compiler version required to 0.8.20 ([#4288](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4288))
+- Bumped minimum compiler version required to 0.8.20 ([#4489](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4489))
```
Comment (Amxx) [inline CHANGELOG.md:51]:
```suggestion
- Bumped minimum compiler version required to 0.8.20 ([#4288](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4288), [#4489](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4489))
```

**Review by Amxx — APPROVED** [review]

**Review by ernestognw — APPROVED** [review]

**Pull request merged**